### PR TITLE
add books3

### PR DESCRIPTION
**Pull request by richarddwang**
books3 is part of EleutherAI/The Pile, but AFAIK, The Pile dataset blend all sub datasets together thus we are not able to use just one of its sub dataset from The Pile data. So I create an independent dataset using The Pile preliminary components.

When I was creating dataset card. I found there is room for creating / editing dataset card. I've made it an issue. #2797 

Also I am wondering whether the import of The Pile dataset is actively undertaken (because I may need it recently)? #1675

### CODE REVIEW

File: datasets/books3/README.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+- found
+languages:
+- en
+licenses:
+- mit
+multilinguality:
+- monolingual
+pretty_name: Books3
+size_categories:
+- unknown
```
Comment (lhoestq) [inline datasets/books3/README.md:14]:
```suggestion
- 100K<n<1M
```

File: datasets/books3/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+This dataset is Shawn Presser's work and is part of EleutherAi/The Pile dataset. 
+
+This dataset contains all of bibliotik in plain .txt form, aka 197,000 books processed in exactly  the same way as did for bookcorpusopen (a.k.a. books1). seems to be similar to OpenAI's mysterious  "books2" dataset referenced in their papers. Unfortunately OpenAI will not give details, so we know very little about any differences. People suspect it's "all of libgen", but it's purely conjecture.
+
+|download_size|36.8 Gib|
+|dataset_size|100.9 Gib|
+
+### Supported Tasks and Leaderboards
+
+- `lm`
```
Comment (lhoestq) [inline datasets/books3/README.md:75]:
```suggestion
This dataset is used for Language Modeling.
```

File: datasets/books3/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+### Languages
+
+- `en`
```
Comment (lhoestq) [inline datasets/books3/README.md:79]:
```suggestion
The dataset is in English.
```

File: datasets/books3/README.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+```
+{'title': '07 LEGO Ninjago - The Search For Zane (Scholastic) - Kate Howard (retail)'
+'text': '\n\nTITLE PAGE\n\nFROM THE JOURNAL OF SENSEI GARMADON\n\nCHAPTER 1\n\nCHAPTER 2\n\nCHAPTER 3\n\nCHAPTER 4\n\nCHAPTER 5\n\nCHAPTER 6\n\nCHAPTER 7\n\nCHAPTER 8\n\nCHAPTER 9\n\nCOPYRIGHT\n\nThroughout Ninjago", five ninja are well-known for their speed, strength, and  of course  the elemental powers that help them protect our world from evil. But there are others who possess some of the same powers as the ninja. Others who may not always use their powers for good.\n\nBefore now, the ninja believed they were special. They di.......'}
+```
+
+### Data Fields
+
+- `title`
+- `text`
```
Comment (lhoestq) [inline datasets/books3/README.md:93]:
```suggestion
- `title`: title of the book
- `text`: text content of the book
```

File: datasets/books3/README.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    - [Dataset Curators](#dataset-curators)
+    - [Licensing Information](#licensing-information)
+    - [Citation Information](#citation-information)
+    - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** https://github.com/soskek/bookcorpus/issues/27#issuecomment-716104208
+- **Repository:** [Needs More Information]
+- **Paper:** https://arxiv.org/abs/2101.00027
```
Comment (lhoestq) [inline datasets/books3/README.md:60]:
```suggestion
- **Homepage:** [GitHub](https://github.com/soskek/bookcorpus/issues/27#issuecomment-716104208)
- **Repository:** [Needs More Information]
- **Paper:** [arXiv](https://arxiv.org/abs/2101.00027)
```

**Review by lhoestq — APPROVED** [review]
Awesome thank you ! :D 

Also note that dataset streaming won't work on this dataset because:
- it is stored as a TAR archive, which is not easily streamable
- we haven't added support for pathlib.Path manipulations to navigate in remote archives

**Comment by lhoestq** [comment]
> When I was creating dataset card. I found there is room for creating / editing dataset card. I've made it an issue. #2797

Thanks for the message, we'll definitely improve this

> Also I am wondering whether the import of The Pile dataset is actively undertaken (because I may need it recently)? #1675

Well currently no, but I think @lewtun was about to do it (though he's currently on vacations)

**Comment by lewtun** [comment]
> > Also I am wondering whether the import of The Pile dataset is actively undertaken (because I may need it recently)? #1675
> 
> Well currently no, but I think @lewtun was about to do it (though he's currently on vacations)

yes i plan to start working on this next week #2185 

one question for @richarddwang - do you know if eleutherai happened to also release the "existing" datasets like enron emails and opensubtitles? 

in appendix c of their paper, they provide details on how they extracted these datasets, but it would be nice if we could just point to a url so we can be as close as possible to original implementation.

**Comment by richarddwang** [comment]
@lewtun 
Sadly, I don't know any existing dataset of enron emails, but I believe opensubtitles dataset should be hosted at here. https://the-eye.eu/public/AI/pile_preliminary_components/
image

**Comment by lewtun** [comment]
thanks for the link @richarddwang! i think that corpus is actually the youtube subtitles one and my impression is that eleutherai have only uploaded the 14 new datasets they created. i've contacted one of the authors so hopefully they can share some additional info for us :)

btw it might take a while to put together all the corpora if i also need to preprocess them (e.g. the open subtitles / enron email etc), but i expect no longer than a few weeks.

**Pull request merged**